plugins/woocommerce/includes/import/class-wc-product-csv-importer.php

```diff
@@ -590,6 +590,35 @@ public function parse_date_field( $value ) {
 		return null;
 	}
 
+	/**
+	 * Parse dates from a CSV.
+	 * Dates can be Unix timestamps or in any format supported by strtotime().
+	 *
+	 * @param string $value Field value.
+	 *
+	 * @return string|null
+	 */
+	public function parse_datetime_field( $value ) {
+		try {
+			// If value is a Unix timestamp, convert it to a datetime string.
+			if ( is_numeric( $value ) ) {
+				$datetime = new DateTime( "@{$value}" );
+				// Return datetime string in ISO8601 format (eg. 2018-01-01T00:00:00Z) to preserve UTC timezone since Unix timestamps are always UTC.
+				return $datetime->format( 'Y-m-d\TH:i:s\Z' );
+			}
+			// Check whether the value is a valid date string.
+			if (false !== strtotime( $value ) ) {
+				// If the value is a valid date string, return as is.
+				return $value;
+			}
+			// If value is not valid Unix timestamp or date string, return null.
```

Comment:
Could you move this return to after the try/catch block so the funtion always falls back to explicitly returning a `null`?

Reply:
There is also one code sniff issue that needs to be addressed.